pkg/detector/detector.go

```diff
@@ -320,6 +321,36 @@ func (d *ResourceDetector) OnAdd(obj interface{}) {
 
 // OnUpdate handles object update event and push the object to queue.
 func (d *ResourceDetector) OnUpdate(oldObj, newObj interface{}) {
+	unstructuredOldObj, err := helper.ToUnstructured(oldObj)
+	if err != nil {
+		klog.Errorf("Failed to transform oldObj, error: %v", err)
+		d.OnAdd(newObj)
```

Comment:
Have you ever occurs a converting error?

Reply:
> Have you ever occurs a converting error?

No, but I referred to pkg\clusterdiscovery\clusterapi.go:153.

Reply:
If so, when we occur err, maybe we need not to call `OnAdd` method.

Reply:
Ok, I was thinking about not changing the original processing logic when the transform fails because it might be processed elsewhere, but if it's not necessnary, I'll remove it.